### ISSUE TITLE
Using YOLOv9c and YOLOv9e to train the hrsc2016 dataset has the same results

### ISSUE THREAD

**Issue by Wei0209** [ultralytics/ultralytics#21063]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

When I use YOLOv9c and YOLOv9e to train the hrsc2016 dataset, the mAP50 values ​​displayed are the same.
I can provide yolov9e.yaml and yolov9c.yaml
# YOLOv9

# parameters
nc: 1  # number of classes

# gelan backbone
backbone:
  - [-1, 1, Conv, [64, 3, 2]]  # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]]  # 1-P2/4
  - [-1, 1, RepNCSPELAN4, [256, 128, 64, 1]]  # 2
  - [-1, 1, ADown, [256]]  # 3-P3/8
  - [-1, 1, RepNCSPELAN4, [512, 256, 128, 1]]  # 4
  - [-1, 1, ADown, [512]]  # 5-P4/16
  - [-1, 1, RepNCSPELAN4, [512, 512, 256, 1]]  # 6
  - [-1, 1, ADown, [512]]  # 7-P5/32
  - [-1, 1, RepNCSPELAN4, [512, 512, 256, 1]]  # 8
  - [-1, 1, SPPELAN, [512, 256]]  # 9

head:
  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 6], 1, Concat, [1]]  # cat backbone P4
  - [-1, 1, RepNCSPELAN4, [512, 512, 256, 1]]  # 12

  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 4], 1, Concat, [1]]  # cat backbone P3
  - [-1, 1, RepNCSPELAN4, [256, 256, 128, 1]]  # 15 (P3/8-small)

  - [-1, 1, ADown, [256]]
  - [[-1, 12], 1, Concat, [1]]  # cat head P4
  - [-1, 1, RepNCSPELAN4, [512, 512, 256, 1]]  # 18 (P4/16-medium)

  - [-1, 1, ADown, [512]]
  - [[-1, 9], 1, Concat, [1]]  # cat head P5
  - [-1, 1, RepNCSPELAN4, [512, 512, 256, 1]]  # 21 (P5/32-large)

  - [[15, 18, 21], 1, OBB, [nc, 1]]  # DDetect(P3, P4, P5)

# YOLOv9

# parameters
nc: 1  # number of classes

# gelan backbone
backbone:
  - [-1, 1, Silence, []]
  - [-1, 1, Conv, [64, 3, 2]]  # 1-P1/2
  - [-1, 1, Conv, [128, 3, 2]]  # 2-P2/4
  - [-1, 1, RepNCSPELAN4, [256, 128, 64, 2]]  # 3
  - [-1, 1, ADown, [256]]  # 4-P3/8
  - [-1, 1, RepNCSPELAN4, [512, 256, 128, 2]]  # 5
  - [-1, 1, ADown, [512]]  # 6-P4/16
  - [-1, 1, RepNCSPELAN4, [1024, 512, 256, 2]]  # 7
  - [-1, 1, ADown, [1024]]  # 8-P5/32
  - [-1, 1, RepNCSPELAN4, [1024, 512, 256, 2]]  # 9
   
  - [1, 1, CBLinear, [[64]]] # 10
  - [3, 1, CBLinear, [[64, 128]]] # 11
  - [5, 1, CBLinear, [[64, 128, 256]]] # 12
  - [7, 1, CBLinear, [[64, 128, 256, 512]]] # 13
  - [9, 1, CBLinear, [[64, 128, 256, 512, 1024]]] # 14
  
  - [0, 1, Conv, [64, 3, 2]]  # 15-P1/2
  - [[10, 11, 12, 13, 14, -1], 1, CBFuse, [[0, 0, 0, 0, 0]]] # 16
  - [-1, 1, Conv, [128, 3, 2]]  # 17-P2/4
  - [[11, 12, 13, 14, -1], 1, CBFuse, [[1, 1, 1, 1]]] # 18  
  - [-1, 1, RepNCSPELAN4, [256, 128, 64, 2]]  # 19
  - [-1, 1, ADown, [256]]  # 20-P3/8
  - [[12, 13, 14, -1], 1, CBFuse, [[2, 2, 2]]] # 21  
  - [-1, 1, RepNCSPELAN4, [512, 256, 128, 2]]  # 22
  - [-1, 1, ADown, [512]]  # 23-P4/16
  - [[13, 14, -1], 1, CBFuse, [[3, 3]]] # 24 
  - [-1, 1, RepNCSPELAN4, [1024, 512, 256, 2]]  # 25
  - [-1, 1, ADown, [1024]]  # 26-P5/32
  - [[14, -1], 1, CBFuse, [[4]]] # 27
  - [-1, 1, RepNCSPELAN4, [1024, 512, 256, 2]]  # 28
  - [-1, 1, SPPELAN, [512, 256]]  # 29

# gelan head
head:
  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 25], 1, Concat, [1]]  # cat backbone P4
  - [-1, 1, RepNCSPELAN4, [512, 512, 256, 2]]  # 32

  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 22], 1, Concat, [1]]  # cat backbone P3
  - [-1, 1, RepNCSPELAN4, [256, 256, 128, 2]]  # 35 (P3/8-small)

  - [-1, 1, ADown, [256]]
  - [[-1, 32], 1, Concat, [1]]  # cat head P4
  - [-1, 1, RepNCSPELAN4, [512, 512, 256, 2]]  # 38 (P4/16-medium)

  - [-1, 1, ADown, [512]]
  - [[-1, 29], 1, Concat, [1]]  # cat head P5
  - [-1, 1, RepNCSPELAN4, [512, 1024, 512, 2]]  # 41 (P5/32-large)

   # detect
  - [[35, 38, 41], 1, OBB, [nc,2]]  # Detect(P3, P4, P5)

Execute the command:
train yolov9c:

from ultralytics import YOLO
 
def main():
    model = YOLO('/workspace/yolov10/ultralytics/cfg/models/v9/yolov9c.yaml',task="obb").load('/workspace/yolov10/yolov9c.pt')  
    model.train(data='/workspace/yolov10/ultralytics/cfg/datasets/hrsc.yaml', optimizer="SGD", epochs=150, imgsz=640, batch=32,device=[0,1,2,3])
if __name__ == '__main__':
    main()

train yolov9e:

from ultralytics import YOLO
 
def main():
    model = YOLO('/workspace/yolov10/ultralytics/cfg/models/v9/yolov9e.yaml',task="obb").load('/workspace/yolov10/yolov9e.pt')  
    model.train(data='/workspace/yolov10/ultralytics/cfg/datasets/hrsc.yaml', optimizer="SGD", epochs=150, imgsz=640, batch=32,device=[0,1,2,3])
if __name__ == '__main__':
    main()

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Wei0209, thank you for reaching out and sharing your experience with YOLOv9c and YOLOv9e on the hrsc2016 dataset! 🚀 This is an automated response to help get you started—an Ultralytics engineer will assist you soon.

We recommend reviewing the [Docs](https://docs.ultralytics.com/) for insights into model configuration and training. If you haven't already, please check out the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage guides for helpful examples.

Since this may be a bug or unexpected behavior, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/)? This will help us investigate further. Key details to include:
- The exact YAML files you used for both models
- Any relevant training logs or output
- Clarification if you meant to use `yolov9e.yaml` but referenced `yolov9c.yaml` in both training scripts

If you have questions about custom training, please share as much information as possible (such as dataset samples and logs), and review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For live support, join our [Discord](https://discord.com/invite/ultralytics) 🎧, or discuss in-depth on [Discourse](https://community.ultralytics.com/). You can also connect with others on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

To ensure you’re using the latest features and fixes, upgrade `ultralytics` and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be run in any of these up-to-date, verified environments (with all dependencies, including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for your detailed report! An Ultralytics engineer will review and follow up with you soon.

**Comment by Wei0209** [comment]
train yolov9c on hrsc:
```python
from ultralytics import YOLO
 
def main():
    model = YOLO('/workspace/yolov10/ultralytics/cfg/models/v9/yolov9c.yaml',task="obb").load('/workspace/yolov10/yolov9c.pt')
    model.train(data='/workspace/yolov10/ultralytics/cfg/datasets/hrsc.yaml', optimizer="SGD", epochs=150, imgsz=640, batch=32,device=[0,1,2,3]  )
if __name__ == '__main__':
    main()
```

train yolov9e on hrsc:
```python
from ultralytics import YOLO
 
def main():
    model = YOLO('/workspace/yolov10/ultralytics/cfg/models/v9/yolov9e.yaml',task="obb").load('/workspace/yolov10/yolov9e.pt')
    model.train(data='/workspace/yolov10/ultralytics/cfg/datasets/hrsc.yaml', optimizer="SGD", epochs=150, imgsz=640, batch=32,device=[0,1,2,3]  )
if __name__ == '__main__':
    main()
```

**Comment by glenn-jocher** [comment]
The identical mAP50 results between YOLOv9c and YOLOv9e suggest there may be an issue with model loading or configuration. YOLOv9e should significantly outperform YOLOv9c based on the [performance benchmarks](https://docs.ultralytics.com/models/yolov9/) showing YOLOv9e achieves 55.6 mAP compared to YOLOv9c's 53.0 mAP.

Please verify that your pretrained weights (`yolov9c.pt` and `yolov9e.pt`) are actually different models and not the same file. You can check this by printing the model architecture or parameter count after loading:

```python
model = YOLO('/workspace/yolov10/ultralytics/cfg/models/v9/yolov9e.yaml',task="obb").load('/workspace/yolov10/yolov9e.pt')
print(f"Model parameters: {sum(p.numel() for p in model.model.parameters())}")
```

Also try training from scratch without loading pretrained weights to see if the architectures produce different results. What specific mAP50 values are you seeing for both models?

**Comment by Wei0209** [comment]
```python
from ultralytics import YOLO
model_c = YOLO('/workspace/yolov10/ultralytics/cfg/models/v9/yolov9c.yaml',task="obb").load('/workspace/yolov10/yolov9c.pt')
model_e = YOLO('/workspace/yolov10/ultralytics/cfg/models/v9/yolov9e.yaml',task="obb").load('/workspace/yolov10/yolov9e.pt')
print(f"Model_c parameters: {sum(p.numel() for p in model_c.model.parameters())}")
print(f"Model_e parameters: {sum(p.numel() for p in model_e.model.parameters())}")
```
Results:
root@d9cba2be85ef:/workspace/yolov10# python3 see_pt.py 
Transferred 921/937 items from pretrained weights
Transferred 1789/1811 items from pretrained weights
Model_c parameters: 36550845
Model_e parameters: 95118781

I will try training from scratch without loading pre-trained weights.

**Comment by Wei0209** [comment]
我嘗試使用hrsc2016資料集從零開始訓練yolov9c與yolov9e模型的結果為:
yolov9c:mAP50: 0.965 mAP5095: 0.824
yolov9e:mAP50: 0.956 mAP5095: 0.817
我認為非常不合理，請問原因為何?

```python
from ultralytics import YOLO
 def main():
    model = YOLO('/ultralytics/ultralytics/cfg/models/v9/yolov9e.yaml',task="obb")
    model = YOLO("/ultralytics/yolov9e.pt")
    model = YOLO("/ultralytics/ultralytics/cfg/models/v9/yolov9e.yaml").load("/ultralytics/yolov9e.pt")
    result = model.train(data='/ultralytics/ultralytics/cfg/datasets/hrsc.yaml', optimizer="SGD", epochs=300, imgsz=640, batch=16,device=[0,1,2,3])
if __name__ == '__main__':
    main()
```

```python
from ultralytics import YOLO
def main():
    model = YOLO('/ultralytics/ultralytics/cfg/models/v9/yolov9c.yaml',task="obb")
    model = YOLO("/ultralytics/yolov9c.pt")
    model = YOLO("/ultralytics/ultralytics/cfg/models/v9/yolov9c.yaml").load("/ultralytics/yolov9c.pt")
    result = model.train(data='/ultralytics/ultralytics/cfg/datasets/hrsc.yaml', optimizer="SGD", epochs=300, imgsz=640, batch=16,device=[0,1,2,3])
if __name__ == '__main__':
    main()
```

**Comment by glenn-jocher** [comment]
The issue is in your training code - you're still loading pretrained weights even when trying to train from scratch. Your code reassigns the model variable multiple times, ultimately loading the pretrained weights:

```python
model = YOLO('/ultralytics/ultralytics/cfg/models/v9/yolov9e.yaml',task="obb")  # from scratch
model = YOLO("/ultralytics/yolov9e.pt")  # overwrites with pretrained
model = YOLO("/ultralytics/ultralytics/cfg/models/v9/yolov9e.yaml").load("/ultralytics/yolov9e.pt")  # loads pretrained again
```

For true from-scratch training, use only:
```python
model = YOLO('/ultralytics/ultralytics/cfg/models/v9/yolov9e.yaml', task="obb")
```

The very high mAP scores (0.965, 0.956) suggest potential overfitting on the small HRSC2016 dataset. With limited data, YOLOv9c's smaller capacity might actually perform slightly better than YOLOv9e since larger models are more prone to overfitting on small datasets. Try reducing the training epochs, adding data augmentation, or using early stopping to prevent overfitting.

**Comment by Wei0209** [comment]
Assuming my task is OBB, when I use model.val(), which iou will the system preset for me to use? Can the data calculated by this iou be fairly compared with other existing methods? For example, will it be looser than piou or dota devkit?

**Comment by glenn-jocher** [comment]
For OBB tasks, Ultralytics uses rotated IoU calculation by default during validation, which properly handles the orientation of bounding boxes. The default IoU thresholds are 0.5 for mAP50 and 0.5:0.95 for mAP50-95, consistent with COCO evaluation standards.

This rotated IoU should be comparable to other standard OBB evaluation methods, but there can be subtle implementation differences between frameworks. For fair comparison with published results, ensure you're using the same evaluation protocol as the reference methods. If you need to match specific evaluation standards like DOTA devkit, you may want to verify the IoU calculation implementation or use the same evaluation toolkit for consistency.

You can check the current IoU calculation by examining the validation metrics output, which will show the specific IoU method being used in the results summary.
